src/blocked.c

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright Redis Ltd. 2023 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "redisraft.h"
+
+#include <math.h>
+#include <string.h>
+#include <strings.h>
+
+/* A list of all current blocked commands */
+
+BlockedCommand *allocBlockedCommand(const char *cmd_name, raft_index_t idx, raft_session_t session, const char *data, size_t data_len, RaftReq *req, RedisModuleCallReply *reply)
+{
+    BlockedCommand *bc = RedisModule_Calloc(1, sizeof(BlockedCommand));
+    bc->command = RedisModule_Strdup(cmd_name);
```

Comment:
Is this guaranteed to be null terminated? As far as I can see you get it from RedisModuleString which may not have null terminator.

Reply:
so my experience is that command names are actually null terminated C strings, its things that can be binary (i.e. keys/values) that are not C strings and hence length matters